### PR TITLE
Add reset time to rate limit exception

### PR DESCRIPTION
**Pull request by jaybyrrd**
This pull request handles to things:

1. Adds reset time to rate limit exception so that developers have the time they need to wait until.
2. Fixes an issue that wasn't preventing compilation where we are assigning a Helix Stream to a v5 stream. 

In order to avoid conflicts I have already merged Dev into this branch, which is where you see quite a few other changes.

### CODE REVIEW

**Comment by jaybyrrd** [comment]
It may be worth refactoring Stream to TwitchStream so that it doesn't conflict with System.IO.Stream, but I am not sure if that is what we want.

File: TwitchLib.Api/Exceptions/TooManyRequestsException.cs [hunk 1]
```diff
@@ -4,13 +4,29 @@ namespace TwitchLib.Api.Exceptions
...
+        /// <inheritdoc />
+        /// <summary>
+        /// Constructor that allows a reset time to be added
+        /// </summary>
+        /// <param name="data">This is the exception message as a string</param>
+        /// <param name="resetTime">This is the reset time from twitch as a linux timestamp</param>
+        public TooManyRequestsException(string data, string resetTime)
+            : base(data)
+        {
+            if(Double.TryParse(resetTime, out var time))
```
Comment (RRickkert) [inline TwitchLib.Api/Exceptions/TooManyRequestsException.cs]: Ew. Double with a capital D:

Comment (jaybyrrd) [reply TwitchLib.Api/Exceptions/TooManyRequestsException.cs]: Made it the primitive parse! :)

File: TwitchLib.Api.Models/Helix/Entitlements/CreateEntitlementGrantsUploadURL/CreateEntitlementGrantsUploadURLResponse.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 using System.Collections.Generic;
 using System.Text;
 
-namespace TwitchLib.Api.Models.Helix.Entitlements
+namespace TwitchLib.Api.Models.Helix.Entitlements.CreateEntitlementGrantsUploadURL
```
Comment (LuckyNoS7evin) [inline TwitchLib.Api.Models/Helix/Entitlements/CreateEntitlementGrantsUploadURL/CreateEntitlementGrantsUploadURLResponse.cs]: Why are we adding an extra namespace here? and in other places. They are not needed.

File: TwitchLib.Api/Exceptions/TooManyRequestsException.cs [hunk 3]
```diff
@@ -4,13 +4,29 @@ namespace TwitchLib.Api.Exceptions
 {
     /// <inheritdoc />
     /// <summary>Exception representing a 429 Http Statuscode</summary>
-    public class TooManyRequestsException : Exception
+    public sealed class TooManyRequestsException : Exception
```
Comment (LuckyNoS7evin) [inline TwitchLib.Api/Exceptions/TooManyRequestsException.cs]: nice, I don't use sealed to often but generally a good use for one here

File: TwitchLib.Api/Sections/Entitlements.cs [hunk 4]
```diff
@@ -19,12 +21,13 @@ public HelixApi(TwitchAPI api) : base(api)
...
             #region CreateEntitlementGrantsUploadURL
-            public Task<Models.Helix.Entitlements.CreateEntitlementGrantsUploadURLResponse> CreateEntitlementGrantsUploadURL(string manifestId, Enums.EntitleGrantType type, string url = null, string applicationAccessToken = null)
+            public Task<Models.Helix.Entitlements.CreateEntitlementGrantsUploadURL.CreateEntitlementGrantsUploadURLResponse> CreateEntitlementGrantsUploadURL(string manifestId, Enums.EntitleGrantType type, string url = null, string applicationAccessToken = null)
             {
                 if (manifestId == null)
                     throw new BadParameterException("manifestId cannot be null");
 
-                var getParams = new List<KeyValuePair<string, string>>() { new KeyValuePair<string, string>("manifest_id", manifestId) };
+                var getParams = new List<KeyValuePair<string, string>>();
+                getParams.Add(new KeyValuePair<string, string>("manifest_id", manifestId));
```
Comment (LuckyNoS7evin) [inline TwitchLib.Api/Sections/Entitlements.cs]:
```
var getParams = new List<KeyValuePair<string, string>>()
{ 
            new KeyValuePair<string, string>("manifest_id", manifestId) 
};
```
I feel this looks better to be honest, but only suggested change

File: TwitchLib.Api/Sections/Entitlements.cs [hunk 5]
```diff
@@ -34,7 +37,7 @@ public Task<Models.Helix.Entitlements.CreateEntitlementGrantsUploadURLResponse>
                         throw new BadParameterException("Unknown entitlement grant type");
                 }
 
-                return Api.TwitchGetGenericAsync<Models.Helix.Entitlements.CreateEntitlementGrantsUploadURLResponse>("/entitlements/upload", ApiVersion.Helix, getParams);
+                return Api.TwitchGetGenericAsync<Models.Helix.Entitlements.CreateEntitlementGrantsUploadURL.CreateEntitlementGrantsUploadURLResponse>("/entitlements/upload", ApiVersion.Helix, getParams);
```
Comment (LuckyNoS7evin) [inline TwitchLib.Api/Sections/Entitlements.cs]: side effect of the added namespace values

**Comment by jaybyrrd** [comment]
I am going to kill this pull request and make the changes fresh off of Dev

**Pull request closed without merging**